firestore/testapp/Assets/Firebase/Sample/Firestore/UIHandlerAutomated.cs

```diff
@@ -1468,15 +1707,76 @@ private class TestException : Exception {
           AssertTaskSucceeds(task);
         }
 
-        DocumentReference doc = db.Document(documentPath);
+        DocumentReference doc =TestFirestore().Document(documentPath);
         DocumentSnapshot snapshot = AssertTaskSucceeds(doc.GetSnapshotAsync(Source.Server));
         int actualValue = snapshot.GetValue<int>("count", ServerTimestampBehavior.None);
         int expectedValue = numTransactionsPerFirestore * firestores.Length;
         AssertEq<int>(actualValue, expectedValue);
 
         foreach (FirebaseApp app in apps) {
-          app.Dispose();
+          DeleteApp(app);
+        }
+      });
+    }
+
+
+    Task TestTransactionsFromMultiDBInParallel() {
+      return Async(() => {
+        // TODO(b/284877917): Remove the Firestore emulator check when multi-db is supported
+        // by production.
+        if (!IsFirestoreEmulatorAvailable()) {
+          return;
+        }
+        FirebaseApp app =  FirebaseApp.Create(FirebaseApp.DefaultInstance.Options, "multi-DB-transactions-in-parallel");
+
+        int numOfFirestores = 3;
+        int numTransactionsPerFirestore = 3;
+
+        FirebaseFirestore[] firestores = new FirebaseFirestore[3];
+        DocumentReference[] docs = new DocumentReference[3];
+        for (int i = 0; i < numOfFirestores; i++)
+        {
+          FirebaseFirestore firestore = TestCustomFirestore(app, "transactions-in-parallel" + (i));
+          firestores[i] = firestore;
+          docs[i] = firestore.Collection("foo").Document();
+        }
+        
+        List<Task> tasks = new List<Task>();
+        for (int i = 0; i < numTransactionsPerFirestore; i++)
+        {
```

Comment:
We need to be consistent with placement of `{`. This happens a few places in PR.